gadget/layout.go

```diff
@@ -233,36 +199,13 @@ func layoutVSFromDiskData(volume *Volume, gadgetToDiskStruct map[int]*OnDiskStru
 
 func layoutVolumeStructures(volume *Volume, gadgetToDiskStruct map[int]*OnDiskStructure) (
 	structures []LaidOutStructure, err error) {
-	// XXX TEMPORARY - next changes will make sure we get always a valid
-	// gadgetToDiskStruct. Remaining cases are calls from ValidateContent
-	// and writeResolvedContentImpl (image build time, initramfs).
-	if len(gadgetToDiskStruct) > 0 {
-		structures, err = layoutVSFromDiskData(volume, gadgetToDiskStruct)
-		if err != nil {
-			return nil, err
-		}
-	} else {
-		structures = layoutVSFromGadget(volume)
+	if len(gadgetToDiskStruct) == 0 {
+		return nil, fmt.Errorf("cannot lay out: internal error: no disk structures provided")
 	}
-
-	// Check:
-	// - No structure overlaps
-	// - offset-write meets the restrictions defined in the gadget
-	// TODO Reuse fully validateCrossVolumeStructure
-	previousEnd := quantity.Offset(0)
-	for idx, ps := range structures {
-		// XXX this check is probably not needed if using matched structures
-		// and will be removed when we always have a gadgetToDiskStruct.
-		if ps.StartOffset < previousEnd {
-			return nil, fmt.Errorf("cannot lay out volume, structure %v overlaps with preceding structure %v", ps, structures[idx-1])
-		}
-		previousEnd = ps.StartOffset + quantity.Offset(ps.Size)
-
-		if err := validateOffsetWrite(ps.VolumeStructure, structures[0].VolumeStructure, volume.MinSize()); err != nil {
-			return nil, err
-		}
+	structures, err = layoutVSFromDiskData(volume, gadgetToDiskStruct)
+	if err != nil {
+		return nil, err
```

Comment:
Can this be simplified?
```suggestion
	return layoutVSFromDiskData(volume, gadgetToDiskStruct)
```

Reply:
Indeed, done now